common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -105,12 +106,17 @@ def may_certify_for_course(certificates_display_behavior, certificates_show_befo
         certificates_show_before_end (bool): whether user can download the
             course's certificates before the course has ended.
         has_ended (bool): Whether the course has ended.
+        certificate_available_date (datetime): the date the certificate is available on for the course.
     """
     show_early = (
         certificates_display_behavior in ('early_with_info', 'early_no_info')
         or certificates_show_before_end
     )
-    return show_early or has_ended
+    past_availability_date = (
+        certificate_available_date
+        and certificate_available_date < datetime.now(utc)
+    )
+    return past_availability_date or show_early or has_ended
```

Comment:
Oh, hm..I hadn't thought about what should happen if the certificate availability date had not yet passed (2 days after course end) but `show_early` was set to end. I think I assumed that `early_with_info`, `early_no_info`, and `end` all took precedence over the certificate availability date. I can clarify with ... Shelby? Or Greg?